Automattic/wp-calypso
# Launchpad: Use new task definition and tracking on the setup free flow

Fix part of #85694

## Proposed Changes

* Transform the free flow tasks to links. 
* Refactor the getEnhancedTasks to from task-helper to remove task definitions
* Move all flow `free` definitions to use the code structure.
* Improve the user tracking on the new task definitions
* Unify the Task type definition
* [Adopt strangler fig pattern using feature flag to migrate gradually all task definitions ](https://dennylesmana.medium.com/what-is-the-strangler-fig-pattern-1560443b8459#:~:text=The%20Strangler%20Fig%20pattern%20aims,that%20isn't%20used%20yet)
* The remaining flow will be covered by other PRs (way smaller than this one).

---
### Technical Notes
As part of my task I moved the big switch that was processing the tasks to the function getDeprecatedTaskDefinition  
(https://github.com/Automattic/wp-calypso/pull/86293/files#diff-b088aae90940dec602186724a7f297ec5e369e75cf503e9cdc41816892b2b628R19) 
and create a new module `steps-repository/launchpad/task-definitions/index.ts` with the task definitions divided by domain.  

The overall idea is to migrate all of this task definition by flow, testing them gradually.

Since it is a very sensitive product area and with low test coverage, this change is under a feature flag, so  we are going to remove the old code as soon all tasks were migrated


--- 

## Testing Instructions
* Run the /setup/free on you localmachine or calypso.live (the feature flag is only available on this envs)
* Follow the steps until the launchpad is rendered
* Open the console and check if this messages were rendered 
``` 
using new task definitions plan_selected
task-helper.tsx:249 using new task definitions setup_free
task-helper.tsx:249 using new task definitions design_selected
task-helper.tsx:249 using new task definitions design_edited
task-helper.tsx:249 using new task definitions domain_upsell
task-helper.tsx:249 using new task definitions first_post_published
task-helper.tsx:249 using new task definitions site_launched
``` 
* Check if the all launchpad tasks are links 
* Complete all tasks and launch the site
* Check if the process happened with success
* Check if the events now contains the following attributes:
``` 
blog_id: 227838534 
checklist_completed: false
checklist_slug: free
client: browser
context: fullscreen
environment: development
environment_id: development
flow: free
is_completed: true
order: 3
site_count: 256
site_id_label: wpcom
site_intent: free
site_plan_id: 1
task_id: design_edited
vph: 873
vpw: 2560
``` 




*

## Pre-merge Checklist
